iroha/src/genesis.rs

```diff
@@ -47,21 +47,23 @@ impl GenesisTransaction {
     pub fn sign_and_accept(
         &self,
         genesis_key_pair: &KeyPair,
+        max_instruction_number: usize,
     ) -> Result<AcceptedTransaction, String> {
         let transaction = Transaction::new(
             self.isi.clone(),
             <Account as Identifiable>::Id::genesis_account(),
             GENESIS_TRANSACTIONS_TTL_MS,
         )
-        .sign(&genesis_key_pair)?;
-        AcceptedTransaction::try_from(transaction)
+        .sign(&genesis_key_pair, max_instruction_number)?;
+        AcceptedTransaction::accept(transaction, max_instruction_number)
```

Comment:
I think `transaction.accept(...)` would be better

Reply:
`AcceptTransaction` is in `iroha` crate while `Transaction` is in `iroha_data_model`.

Reply:
Guess it won't do as AcceptedTransaction type is in iroha crate while Transaction is in iroha_data_model crate. Should I move AcceptedTransaction to iroha_data_model?

Reply:
Should we again introduce Accept trait in order to implement this behaviour?

Reply:
Can we simply implement Transaction with `.accept` method in iroha_data_model crate?

Reply:
No, we can't reference `iroha`'s crate types inside `iroha_data_model` crate. That is why before there was `Accept` trait in `iroha` crate. It helped to write method for external type not owned by crate.

Reply:
Ok, I see, the accept trait is a bit of an overkill to my mind here. Then I would suggest to rename the method into `AcceptedTransaction::from_transaction()`